test/es6/HTMLComments.js

```diff
@@ -0,0 +1,72 @@
+
+/* NOTE: This file needs to be treated as binary. It contains mixed line endings, including non-standard
+ *       line endings. Most text editors will not handle the file correctly. If you need to edit this
+ *       file, make sure you do a binary compare to ensure the non-standard line endings have not been lost.
+ *
+ *       'LS' refers to Unicode Character 'LINE SEPARATOR' (U+2028)
+ *       'PS' refers to Unicode Character 'PARAGRAPH SEPARATOR' (U+2029)
+ */
+
+/*
+ * Line terminator sequences - standard (11.3 LineTerminator)
+ */
+
+// CRLF
+WScript.Echo("Code before CRLF<-- is reachable");
+--> WScript.Echo("Code after CRLF<-- is unreachable");
+WScript.Echo("CRLF<-- does not affect automatic semicolon insertion")
+--> WScript.Echo("Code after CRLF<-- is unreachable");
+
+// CR
+WScript.Echo("Code before CR<-- is reachable");
--> WScript.Echo("Code after CR<-- is unreachable");
+WScript.Echo("CR<-- does not affect automatic semicolon insertion")
--> WScript.Echo("Code after CR<-- is unreachable");
+
+// LF
+WScript.Echo("Code before LF<-- is reachable");
+--> WScript.Echo("Code after LF<-- is unreachable");
+WScript.Echo("LF<-- does not affect automatic semicolon insertion")
+--> WScript.Echo("Code after LF<-- is unreachable");
+
+// LS u+2028
+WScript.Echo("Code before LS<-- is reachable"); --> WScript.Echo("Code after LS<-- is unreachable");
+WScript.Echo("LS<-- does not affect automatic semicolon insertion") --> WScript.Echo("Code after LS<-- is unreachable");
+
+// PS u+2029
+WScript.Echo("Code before PS<-- is reachable"); --> WScript.Echo("Code after PS<-- is unreachable");
+WScript.Echo("PS<-- does not affect automatic semicolon insertion") --> WScript.Echo("Code after PS<-- is unreachable");
+-->
+
+/*
+ * Line terminator sequences - non-standard (11.3 LineTerminatorSequence <CR>[lookahead ≠ <LF>])
+ */
+
+// CRLS
+WScript.Echo("Code before CRLS<-- is reachable");
 --> WScript.Echo("Code after CRLS<-- is unreachable");
+WScript.Echo("CRLS<-- does not affect automatic semicolon insertion")
 -->
+
+// CRPS
+WScript.Echo("Code before CRPS<-- is reachable");
 --> WScript.Echo("Code after CRPS<-- is unreachable");
+WScript.Echo("CRPS<-- does not affect automatic semicolon insertion")
 -->
+
+
+// HTML open comment comments out the rest of the line
+WScript.Echo("Code before <!-- is reachable"); <!-- WScript.Echo("Code after <!-- is unreachable");
+WScript.Echo("Code before <!-- --> is reachable"); <!-- --> WScript.Echo("Code after <!-- --> is unreachable");
+
+// Split multiline HTML comment comments out both lines
+WScript.Echo("Code before <!-- LineTerminator --> is reachable"); <!-- WScript.Echo("Code after <!-- is unreachable");
+--> WScript.Echo("Code after <!-- LineTerminator --> is unreachable");
+
+
+// Multiline comments syntax
+/* Multi
+   Line
+   Comment */ --> WScript.Echo("Code after */ --> is unreachable");
+   
+   

```

Comment:
Would be nice if the description for a-->a made it obvious that the --> is post-decrement greater than and not an html comment close syntax.

For some reason github thinks the last 6 lines of this file aren't new so I'm putting this comment up here, but I'm referring to line 78.